pkg/segment/query/processor/evalcommand_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package processor
+
+import (
+	"testing"
+
+	"github.com/siglens/siglens/pkg/segment/query/iqr"
+	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	toputils "github.com/siglens/siglens/pkg/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_EvalCommand(t *testing.T) {
+
+	boolExpr := &structs.BoolExpr{
+		IsTerminal: true,
+		LeftValue: &structs.ValueExpr{
+			ValueExprMode: structs.VEMNumericExpr,
+			NumericExpr: &structs.NumericExpr{
```

Comment:
Hmm this large BoolExpr is very long and a bit hard to read. Maybe we could use our SPL parser to parse an `eval` query and extract the BoolExpr, so this unit test is more obvious. For that we might also have to have this unit test in a separate `tests` package to avoid an import cycle.

I think you can keep this though; I'm not sure if trying to extract the BoolExpr from the parser output will be easy. You should add a comment above this though giving the equivalent SQL command